File: src/Infraestructure/Class/Espacio.ts
import { ICoordenada } from "../Interfaces/ICoordenada";
import { IEspacio } from "../Interfaces/IEspacio";
import { IVehiculo } from "../Interfaces/IVehiculo";
import { Coordenada } from "./Coordenada";

export class Espacio implements IEspacio {
    public Coordenada: ICoordenada;
    public Ocupado: boolean;
    public OcupadoPor: IVehiculo;
    public Tiempo: Date;

    public constructor(piso: number, espacio: number, ocupado: boolean, OcupadoPor: IVehiculo) {
        this.Coordenada = new Coordenada(piso, espacio);
        this.Ocupado = ocupado;
        this.OcupadoPor = OcupadoPor;
        this.Tiempo = new Date();
    }
}

File: src/main/java/Grupo2/Garage/Infraestructure/Class/Garage/Espacio.java
package Grupo2.Garage.Infraestructure.Class.Garage;

import Grupo2.Garage.Infraestructure.Interfaces.IVehiculo;

import java.util.Date;

public class Espacio {
    public Coordenada Coordenada;
    public boolean Ocupado;
    public IVehiculo OcupadoPor;
    public Date Tiempo;

    public Espacio(Integer piso, Integer espacio, boolean ocupado, IVehiculo OcupadoPor){
        this.Coordenada = new Coordenada(piso, espacio);
        this.Ocupado = ocupado;
        this.OcupadoPor = OcupadoPor;
        this.Tiempo = new Date();
    }
}
